### PR TITLE
[11.0][NEW] base_bank_from_iban: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
This module adds code to bank and fill bank from IBAN in Bank Account

@Tecnativa

### CODE REVIEW

File: base_bank_from_iban/README.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+To use this module, you need to:
+
+#. Go to Partner
+#. Click Bank Account(s) in Sales & Purchases tab
+#. Create/modify IBAN bank account
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/110/10.0
```
Comment (yajo) [inline base_bank_from_iban/README.rst]: 11.0

File: base_bank_from_iban/models/account_journal.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2017 Tecnativa - Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3).
+
+from odoo import api, models
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    @api.multi
```
Comment (yajo) [inline base_bank_from_iban/models/account_journal.py]: Implicit since odoo 10.0

File: base_bank_from_iban/models/res_bank.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2017 Tecnativa - Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3).
+
+from odoo import fields, models
+
+
+class ResBank(models.Model):
+    _inherit = 'res.bank'
+
+    code = fields.Char('Code')
```
Comment (yajo) [inline base_bank_from_iban/models/res_bank.py]: The field string equals the name, so it's implicit. You can remove it.

File: base_bank_from_iban/models/res_partner_bank.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+from odoo import api, models
+from odoo.addons.base_iban.models.res_partner_bank import \
+    normalize_iban, pretty_iban, _map_iban_template
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    @api.multi
+    @api.onchange('acc_number')
```
Comment (yajo) [inline base_bank_from_iban/models/res_partner_bank.py]: According to the code, this should depend on `acc_type` too

File: base_bank_from_iban/models/res_partner_bank.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+from odoo import api, models
+from odoo.addons.base_iban.models.res_partner_bank import \
+    normalize_iban, pretty_iban, _map_iban_template
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    @api.multi
```
Comment (yajo) [inline base_bank_from_iban/models/res_partner_bank.py]: Implicit

**Review by yajo — CHANGES_REQUESTED** [review]
Looks awesome, pal. Just some minor improvements.

File: base_bank_from_iban/models/res_partner_bank.py (hunk 4 above)
Comment (carlosdauden) [reply base_bank_from_iban/models/res_partner_bank.py]: It shouldn't be possible to change acc_type without changing acc_number

Comment (pedrobaeza) [reply base_bank_from_iban/models/res_partner_bank.py]: The problem is that we don't know if `acc_type` is computed before or after calling this onchange, so it's better to include that field as dependency here for not depending on the ORM implementation.

**Comment by carlosdauden** [comment]
@Yajo @pedrobaeza changes done

**Review by yajo — APPROVED** [review]

**Comment by hbrunn** [comment]
did you check https://github.com/OCA/community-data-files/pull/13 ? I think the mechanism there is a bit more general, and what's done here can be very simply made fit

**Comment by pedrobaeza** [comment]
@hbrunn I think this mechanism is more standard (it's even an ISO) and NL can fit also here.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I'm going to merge this one as v11 includes support for this directly and it's a dependency from other modules. @hbrunn you can extend this one for your programatic approach. In Spain, that approach can't be and we need to hardcode on each bank the correspondent code.

**Pull request merged**